apis/v1/gateway_types.go

```diff
@@ -606,6 +601,31 @@ type GatewayTLSConfig struct {
 	Options map[AnnotationKey]AnnotationValue `json:"options,omitempty"`
 }
 
+// GatewayTLSConfig specifies frontend tls configuration for gateway.
+type GatewayTLSConfig struct {
+	// defaultTLS specifies the default client certificate validation configuration
+	// for all Listeners handling HTTPS traffic, unless a per-port configuration
+	// is defined.
+	//
+	// support: Core
+	//
+	// +required
+	// <gateway:experimental>
+	DefaultTLS FrontendTLSValidation `json:"defaultTLS"`
+
+	// tlsPerPort specifies tls configuration assigned per port.
+	// Per port configuration is optional. Once set this configuration overrides
+	// the default configuration for all Listeners handling HTTPS traffic
+	// that match this port.
+	// Each override port requires a unique TLS configuration.
+	//
+	// support: Core
+	//
+	// +optional
+	// <gateway:experimental>
+	TLSPerPort []TLSConfig `json:"tlsPerPort,omitempty"`
```

Comment:
Similar comment to above.
```suggestion
	PerPort []TLSConfig `json:"perPort,omitempty"`
```